main/src/library/File.flix

```diff
@@ -364,4 +364,116 @@ namespace File {
             }
         }
 
+    ///
+    /// Writes `data` to the given file `f`.
+    ///
+    /// Creates file `f` if it does not exist. Overwrites it if it exists.
+    ///
+    /// Returns `true` if the file was created.
+    ///
+    pub def write(f: String, data: t): Result[Bool, String] & Impure with ToString[t] =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.PrintWriter(##java.io.File) as newPrintWriter;
+            import java.io.PrintWriter.println(String);
+            import java.io.PrintWriter.close();
+            import java.io.PrintWriter.checkError();
+
+            let javaFile = newFile(f);
+            let w = newPrintWriter(javaFile);
+
+            println(w, "${data}");
+            close(w);
+
+            Ok(Bool.not(checkError(w)))
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Writes `data` to the given `f`.
+    ///
+    /// Creates `f` if it does not exist. Overwrites it if it exists.
+    ///
+    /// Returns `true` if the file was created.
+    ///
+    pub def writeLines(f: String, data: f[String]): Result[Bool, String] & Impure with Foldable[f] =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.PrintWriter(##java.io.File) as newPrintWriter;
+            import java.io.PrintWriter.println(String);
+            import java.io.PrintWriter.close();
+            import java.io.PrintWriter.checkError();
+
+            let javaFile = newFile(f);
+            let w = newPrintWriter(javaFile);
+
+            Foldable.foldLeft((_, line) -> println(w, line), (), data);
+            close(w);
+
+            Ok(Bool.not(checkError(w)))
```

Comment:
`checkError` isn't directly related to whether the file is created or not. I think we'll first check whether the file exists, remember that, then return `not alreadyExists` at the end.

But `checkError` is still a good idea; if it says there was an error then we should return a `Result.Err`

Reply:
Yes, good solution.
As I understand this now, we return `true` is the file was created and `false`, if the file already existed and is overwritten. Is that correct?

Reply:
That's correct. I think this bit of confusion indicates that we should update the documentation with something like the clearer description you just gave.

Reply:
Good idea